x/auction/types/auction.go

```diff
@@ -15,10 +15,6 @@ func (a *Auction) ValidateBasic() error {
 		return sdkerrors.Wrapf(ErrAuctionIDMustBeNonZero, "id: %d", a.Id)
 	}
 
-	if !a.StartingTokensForSale.IsPositive() {
```

Comment:
I don't think we should remove this from `ValidateBasic`, but if the intent is to avoid a panic then we can front-load this specific check in the `BeginAuction` call like you've done. Once the auction is created, this is still a validation we'd want to confirm on an auction struct.